code/iaas/healthcheck/src/main/java/io/cattle/iaas/healthcheck/service/impl/HealthcheckServiceImpl.java

```diff
@@ -259,17 +261,33 @@ protected void createHealthCheckHostMaps(HealthcheckInstanceType instanceType, l
                 HealthcheckInstance.class, healthInstance.getId());
         List<? extends Host> availableActiveHosts = allocatorDao.getActiveHosts(healthInstance.getAccountId());
         
+        // skip hosts labeled accordingly
+        Iterator<? extends Host> it = availableActiveHosts.iterator();
+        while (it.hasNext()) {
```

Comment:
io.cattle.platform.util.type.CollectionUtils.getNestedValue(Object, String...)` would make this a bit more straight forward.